dev/doc/critical-bugs

```diff
@@ -263,6 +263,18 @@ Conflicts with axioms in library
   GH issue number: none
   risk:
 
+  component: primitive integers
+  summary: primitive integer function div21 had implementations not matching its specification
+  introduced: master (before v8.10 in GH pull request #6914)
+  impacted released versions: none
+  impacted development branches:
+  impacted coqchk versions: none
+  fixed in: TBA
```

Comment:
```suggestion
  fixed in: #10046
```

Reply:
For the record, when I started the "critical-bugs" file, I skipped those from master or beta-releases, i.e. I kept only those released in an official X.Y.Z release (see sentence starting with "There were otherwise several bugs in beta-releases").

So, if this bug is added to the list, more should be morally added on the master- and beta-only bugs which I did not list, and the paragraph on critical bugs in beta-versions updated.

Reply:
I think there is no reason to exclude these bugs from the list, except that it would have taken you much longer to list them all, I suppose.

Reply:
> I think there is no reason to exclude these bugs from the list, except that it would have taken you much longer to list them all, I suppose.

I think people are usually interested in bugs affecting released versions. The point is for people auditing code to know if they can trust proofs being checkd by a given Coq version.

At least, we should make it easy to find only the ones impacting released versions.

Note that an inconsistency on "master" may even be local to one commit in a PR.

Reply:
I would tend to be in favor of not mentioning this bug considering that it did not affect any released version *and* that it's not particularly interesting (it comes to no surprise that an incorrect axiom can lead to a proof of False).

In the contrary, maybe some bugs can be worth remembering, even if they did not affect any released version. For instance #9925 can be a good reminder that one has to be careful with the GC in the VM, particularly when calling back OCaml code.

Reply:
OK

Reply:
Thanks, done so